sxt/algorithm/base/BUILD

```diff
@@ -6,6 +6,17 @@ load(
 sxt_cc_component(
     name = "index_functor",
     with_test = False,
+    deps = [
+        "//sxt/execution/async:future",
```

Comment:
Did the `index_functor` component change? Does it need access to `//sxt/execution/async:future`?